thumbor/handlers/__init__.py

```diff
@@ -128,25 +132,25 @@ def get_image(self):
                 self._error(500)
             return
 
-        if not fetch_result.media.is_valid:
-            if LoaderResult.ERROR_NOT_FOUND in fetch_result.media.errors:
+        if not media.is_valid:
+            if LoaderResult.ERROR_NOT_FOUND in media.errors:
                 self._error(404)
                 return
-            elif LoaderResult.ERROR_UPSTREAM in fetch_result.media.errors:
+            elif LoaderResult.ERROR_UPSTREAM in media.errors:
                 # Return a Bad Gateway status if the error came from upstream
                 self._error(502)
                 return
-            elif LoaderResult.ERROR_TIMEOUT in fetch_result.media.errors:
+            elif LoaderResult.ERROR_TIMEOUT in media.errors:
                 # Return a Gateway Timeout status if upstream timed out (i.e. 599)
                 self._error(504)
                 return
             else:
+                logger.error('[BaseHandler] Media is invalid')
                 self._error(500)
                 return
 
-        normalized = fetch_result.normalized
-        media = fetch_result.media
-        engine = fetch_result.engine
+        normalized = media.normalized
+        engine = media.engine
```

Comment:
I agree, the engine should not be stored in the Media object. But do you need to store the engine at all in the result object? It is already in the context.
What exactly is `normalized`?